plugins/@grouparoo/app-templates/src/source/table/getExamples.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  GetColumnDefinitionsMethod,
+  GetSampleRowsMethod,
+  DataResponseRow,
+  ColumnDefinitionMap,
+  FilterOperation,
+  SimpleAppOptions,
+} from "./pluginMethods";
+
+export interface GetExampleRowsMethod {
+  (argument: {
+    connection: any;
+    appOptions: SimpleAppOptions;
+    tableName: string;
+    columns?: ColumnDefinitionMap;
+    getSampleRows: GetSampleRowsMethod;
+    getColumns: GetColumnDefinitionsMethod;
+  }): Promise<DataResponseRow[]>;
+}
+
+export interface GetColumnExamplesMethod {
+  (argument: {
+    connection: any;
+    appOptions: SimpleAppOptions;
+    tableName: string;
+    getSampleRows: GetSampleRowsMethod;
+    getColumns: GetColumnDefinitionsMethod;
+  }): Promise<
+    Array<{
+      key: string;
+      description?: string;
+      examples?: Array<any>;
+    }>
+  >;
+}
+
+export const getExampleRows: GetExampleRowsMethod = async ({
+  connection,
+  appOptions,
+  tableName,
+  columns,
+  getSampleRows,
+  getColumns,
+}) => {
+  let rows = await getSampleRows({ connection, appOptions, tableName });
+  if (!rows || rows.length === 0) {
+    if (!columns) {
+      columns = await getColumns({ connection, appOptions, tableName });
+    }
+    rows = makeRowsFromColumns(columns);
+  }
+  return rows;
+};
+
+function makeRowsFromColumns(columns: ColumnDefinitionMap): DataResponseRow[] {
```

Comment:
Won't this method null-out all the data passed in?

Reply:
The point of these was to return _something_ for previews even on empty tables.
It only sets keys on `sample` and doesn't affect what's passed in. 
It's the same as https://github.com/grouparoo/grouparoo/blob/3dff7b5b567b2d9289015ed5e79e228edffe67af/plugins/%40grouparoo/bigquery/src/lib/util.ts#L159-L164